### PR TITLE
UCT/IB: Fix mask for SLs

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Fix mask for SLs.

## Why ?

before (SL=15 doesn't exist neither in SLs w/ AR nor in SLs w/o AR):
```
1608108548.820578] [dory03:41355:0]        ib_mlx5.c:795  UCX  ERROR AR=n was requested for SL=auto, but could not select SL without AR on mlx5_2:1, SLs with AR support = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14 }, SLs without AR support = { <none> }
```
after
```
[1608108634.925579] [dory03:41479:0]        ib_mlx5.c:795  UCX  ERROR AR=n was requested for SL=auto, but could not select SL without AR on mlx5_2:1, SLs with AR support = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 }, SLs without AR support = { <none> }
```

## How ?

1. Rename `UCT_IB_SL_MAX` to `UCT_IB_SL_NUM` and use `UCT_IB_SL_NUM` instead.
2. Remove `UCT_IB_SL_INVALID` and use `UCT_IB_SL_NUM` instead.

### CODE REVIEW

**Review by brminich — APPROVED** [review]

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -647,7 +647,7 @@ void uct_ib_iface_fill_ah_attr_from_gid_lid(uct_ib_iface_t *iface, uint16_t lid,
 
     memset(ah_attr, 0, sizeof(*ah_attr));
 
-    ucs_assert(iface->config.sl != UCT_IB_SL_INVALID);
+    ucs_assert(iface->config.sl != UCT_IB_SL_NUM);
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c:650]: sl<NUM  || sl==AUTO

File: src/uct/ib/base/ib_iface.c [hunk 2]
```diff
@@ -1179,7 +1179,7 @@ static void uct_ib_iface_set_path_mtu(uct_ib_iface_t *iface,
 
 uint8_t uct_ib_iface_config_select_sl(const uct_ib_iface_config_t *ib_config)
 {
-    ucs_assert((ib_config->sl <= UCT_IB_SL_MAX) ||
+    ucs_assert((ib_config->sl < UCT_IB_SL_NUM) ||
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c:1182]: if (==AUTO) return 0;
assert(sl < NUM);
return (uint8_t)sl;

File: src/uct/ib/base/ib_iface.c [hunk 3]
```diff
@@ -1244,7 +1244,7 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
     self->config.rx_max_batch       = ucs_min(config->rx.max_batch,
                                               config->rx.queue_len / 4);
     self->config.port_num           = port_num;
-    self->config.sl                 = UCT_IB_SL_INVALID;
+    self->config.sl                 = UCT_IB_SL_NUM;
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c:1247]: add comment /* initialize to invalid value */

File: src/uct/ib/base/ib_iface.c (hunk 1 above)
Comment (dmitrygx) [reply src/uct/ib/base/ib_iface.c]: it should be just `sl < NUM` here

File: src/uct/ib/base/ib_iface.c (hunk 2 above)
Comment (dmitrygx) [reply src/uct/ib/base/ib_iface.c]: done

File: src/uct/ib/base/ib_iface.c (hunk 3 above)
Comment (dmitrygx) [reply src/uct/ib/base/ib_iface.c:1251]: done

**Review by yosefe — COMMENTED** [review]
need gtest to catch this? maybe setting SL to 15?

**Comment by dmitrygx** [comment]
> need gtest to catch this? maybe setting SL to 15?

ok, I'll add the gtest

**Pull request closed without merging**